runtime/compiler/control/HookedByTheJit.cpp

```diff
@@ -4332,14 +4332,18 @@ static void jitStateLogic(J9JITConfig * jitConfig, TR::CompilationInfo * compInf
       }
    else // We will go to STEADY or DEEP_STEADY
       {
-      if (oldState == DEEPSTEADY_STATE ||  // keep the DEEPSTEADY_STATE
+      if ((oldState == DEEPSTEADY_STATE) ||  // keep the DEEPSTEADY_STATE
```

Comment:
I'm fine with the ()s around the && to make the precedence clear but why around the == ?

Reply:
See longer explanation above.  But in general it helps code readability and evolution to make each conditional a clear, separate thing.